data/src/main/java/com/microsoft/azure/kusto/data/HttpClientFactory.java

```diff
@@ -0,0 +1,116 @@
+package com.microsoft.azure.kusto.data;
+
+import org.apache.http.HeaderElement;
+import org.apache.http.HeaderElementIterator;
+import org.apache.http.HttpResponse;
+import org.apache.http.conn.ConnectionKeepAliveStrategy;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.message.BasicHeaderElementIterator;
+import org.apache.http.protocol.HTTP;
+import org.apache.http.protocol.HttpContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A singleton factory of HTTP clients.
+ */
+class HttpClientFactory {
+
+    private static final Logger logger = LoggerFactory.getLogger(HttpClientFactory.class);
```

Comment:
nit: consts should be all upper case

Reply:
We used Google code style, but i renamed the logger with upper case to stay consistent with the rest of the code.